src/formatDate.js

```diff
@@ -8,7 +8,57 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const splitElement = fromFormat[3];
+  const joinElement = toFormat[3];
+  const newDate = date.split(splitElement);
+
+  let formatAgan = [];
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    if (fromFormat[i] === 'YY' || fromFormat[i] === 'YYYY') {
+      formatAgan[0] = newDate[i];
+    }
+
+    if (fromFormat[i] === 'MM') {
+      formatAgan[1] = newDate[i];
+    }
+
+    if (fromFormat[i] === 'DD') {
+      formatAgan[2] = newDate[i];
+    }
+  }
```

Comment:
This loop can be simplified. Instead of using multiple if statements, you could use a map or an object to associate format tokens ('YY', 'MM', 'DD') with their respective indices, and then use this map to build the `formatAgan` array directly.